Sources/MapboxDirections/RouteResponse.swift

```diff
@@ -26,6 +26,49 @@ public struct RouteResponse {
      This property does not persist after encoding and decoding.
      */
     public var created: Date = Date()
+    
+    /**
+     Managed of `RoadClasses` restrictions specified to `RouteOptions.roadClassesToAvoid` which were violated during route calculation.
+     
+     Routing engine may still utilize `RoadClasses` meant to be avoided in cases when routing is impossible otherwise.
+     
+     Resulting array is in the same order as `routes`, showing exact `RoadClasses` restrictions were ignored for each particular route at specific leg/step/intersection. `nil` and empty return arrays correspond to `nil` and empty `routes` array.
+     */
+    public var ignoredRoadClassesAvoidance: [RouteRoadClassesViolations]? {
```

Comment:
Sorry I don't seem to understand the idea. Current feature only detects when RoadClasses were meant to be avoided but still got included into the route(s). What do you mean by adding a new property?

Reply:
I assumed you had included “avoidance” in the name with the expectation that we might need a parallel `ignoredRoadClassesAllowance` property for when `roadClassesToAllow` gets violated. So pushing a property down into each `RoadClassExclusionViolation` would allow us to combine the two properties into one in that future scenario. In other words, I don’t think it’s necessary to name this property so specifically; `roadClassViolations` is adequate.

Reply:
Got it. Renamed the property now. If we would like to expand this feature that way later, we would just extend `RouteRoadClassesViolations` to contain new data.